vochain/scrutinizer/scrutinizer.go

```diff
@@ -271,11 +330,18 @@ func (s *Scrutinizer) Commit(height uint32) error {
 			wg.Done()
 		})
 		wg.Wait()
-		s.indexTxLock.Unlock()
+		s.voteTxLock.Unlock()
 		log.Infof("indexed %d new envelopes, took %s",
 			len(s.voteIndexPool), time.Since(startTime))
-		// Add the envelopes to the countTotalEnvelopes var
-		atomic.AddUint64(&s.countTotalEnvelopes, uint64(len(s.voteIndexPool)))
+		envelopeCount := &indexertypes.CountStore{}
+		if err := s.db.Get(indexertypes.CountStore_Envelopes, envelopeCount); err != nil {
```

Comment:
Cannot we assume this index exist already (was created by scrutinizer initialization) and avoid doing two db operations here?

Reply:
You mean by using updateMatching instead?

Reply:
Yes